libs/wlgen/wlgen/rta.py

```diff
@@ -254,25 +254,27 @@ def _confCustom(self):
             self._log.info('   %s', rtapp_conf)
             ifile = open(rtapp_conf, 'r')
 
+        temp_json = list()
         for line in ifile:
             if '__DURATION__' in line and self.duration is None:
                 raise ValueError('Workload duration not specified')
             for src, target in replacements.iteritems():
                 line = line.replace(src, target)
-            ofile.write(line)
+            temp_json.append(line)
+
+        temp_json = '\n'.join(temp_json)
+        self.rta_profile = json.loads(temp_json)
```

Comment:
Don't like the usage of this attribute... which was introduced explicitly for the PROFILE case...
We should have instead a generic configuration container.... as in the example below...